clients/app/src/views/components/compositions/FanClubInfo/FanClubInfo.tsx

```diff
@@ -11,12 +11,39 @@ import { getReadableNumber } from 'utils/helpers/readableNumber';
 import preventMultiCalls from 'utils/helpers/preventMultiCalls';
 import { ProfileListTypeEnum } from 'views/screens/ProfileList';
 import { IFanClubInfoProps } from './types';
+import buildFanClubMembersStore from 'stores/fanClubMembers';
 
 const FanClubInfo: React.FC<IFanClubInfoProps> = ({ data }) => {
   const { pushTo } = useNavigationWithParams();
   const { colors } = useTheme();
 
-  const isJoined = false;
+  const fanClubMembersStore = React.useMemo(() => buildFanClubMembersStore(), []);
+  const { add: storeAdd, delete: storeDelete } = fanClubMembersStore.useSelector(
+    'add',
+    'delete'
+  );
+
+  const [isJoined, setIsJoined] = React.useState(data.joined || false);
+
+  const onJoinLeavePress = React.useCallback(() => {
+    setIsJoined(!isJoined);
+
+    if (isJoined) {
+      fanClubMembersStore.remove({ fanClubId: data.id });
+    } else {
+      fanClubMembersStore.setFieldValue('fanClubId', data.id);
+      fanClubMembersStore.create();
+    }
+  }, [data.id, isJoined, fanClubMembersStore]);
+
+  React.useEffect(() => {
+    if (isJoined && storeAdd.status === 'error') {
```

Comment:
Isn't it better to write this part like this?

```
if (storeAdd.status === 'error')
  setIsJoined(!isJoined);
```

Reply:
changed will push soon!